### PR TITLE
Single CBGT manager per SG instance

### PR DESCRIPTION
**Pull request by tleyden**
Only cbgt.RegisterPIndexImplType once during server context init

https://github.com/couchbase/sync_gateway/issues/1087

### CODE REVIEW

**Comment by tleyden** [comment]
Before merging this, first merge:
- https://github.com/couchbase/sg-bucket/pull/6
- https://github.com/couchbaselabs/walrus/pull/12

File: src/github.com/couchbase/sync_gateway/base/sgw_pindex.go [hunk 1]
```diff
@@ -21,11 +20,25 @@ type CbgtContext struct {
...
+// A map from bucket name -> TapEvent channels that CBGT can write to
+// Somewhat awkward workaround to the face that there's no easy way to get scope
+// on such a channel created in the CouchbaseBucket.StartTapFeed() method.
+// TODO: not sure if this needs to be goroutine safe, assuming it does not for now.
+var TapEventWritableChannels map[string]chan<- sgbucket.TapEvent
+
+func init() {
+	TapEventWritableChannels = make(map[string]chan<- sgbucket.TapEvent)
+}
+
```
Comment (adamcfraser) [inline src/github.com/couchbase/sync_gateway/base/sgw_pindex.go]: Are these obsolete with the new approach?

Comment (tleyden) [inline src/github.com/couchbase/sync_gateway/base/sgw_pindex.go]: Yup will remove

**Comment by tleyden** [comment]
@adamcfraser ok, removed that and pushed -f -- should be ready to merge now.

**Pull request merged**